foxai/metrics/metrics.py

```diff
@@ -0,0 +1,142 @@
+        attributes_map = torch.from_numpy(attributes_map).permute(2,1,0).to(device)
+
+        attributes_map_inv:np.ndarray = np.expand_dims(np.where(preprocessed_attrs <= val, 0, 1), axis=-1)
+        attributes_map_inv = attributes_map_inv.repeat(3, axis=-1)
+        attributes_map_inv = torch.from_numpy(attributes_map_inv).permute(2,1,0).to(device)
+        
+        if metric_type == Metrics.DELETION:
+            perturbed_img:torch.tensor = transformed_img * attributes_map + removed_img_part*attributes_map_inv
+        else:
+             perturbed_img:torch.tensor = removed_img_part * attributes_map + transformed_img*attributes_map_inv
+        
+        perturbed_img = perturbed_img.to(device)
+                
+        output = model(perturbed_img.unsqueeze(dim=0))
+        softmax_output:torch.tensor = torch.nn.functional.softmax(output)[0]
+        importance_lst.append(softmax_output[chosen_class].detach().numpy())
+
+    metric:np.ndarray = np.round(np.trapz(importance_lst)/len(importance_lst), 4)
+    
+    return metric, importance_lst  
+
+def deletion(attrs:torch.tensor, transformed_img:torch.tensor, model:torch.nn.Module, chosen_class:int)->Tuple[np.ndarray, List]:
+    """Calculate deletion metric given importance map, image, model and chosen class. 
+
+    Args:
+        attrs: Torch Tensor corresponding to importance map.
+        transformed_img: Torch Tensor corresponding to image. 
+        model: model which we are explaining.
+        chosen_class: index of the class we are creating metric for. 
+
+    Returns:
+        metric: numerical value of chosen metric for given picture and explanation. 
+        importance_lst: list of numpy elements corresponding to confidence value at each step. 
+
+    Raises:
+        AttributeError: if metric type is not enum of Metrics.INSERTION or Metrics.DELETION
+    """
+    return _metric_calculation(attrs, transformed_img, model, chosen_class, metric_type=Metrics.DELETION)
+
+def insertion(attrs:torch.tensor, transformed_img, model:torch.nn.Module, chosen_class)->Tuple[np.ndarray, List]:
+    """Calculate insertion metric given importance map, image, model and chosen class. 
+
+    Args:
+        attrs: Torch Tensor corresponding to importance map.
+        transformed_img: Torch Tensor corresponding to image. 
+        model: model which we are explaining.
+        chosen_class: index of the class we are creating metric for. 
+
+    Returns:
+        metric: numerical value of chosen metric for given picture and explanation. 
+        importance_lst: list of numpy elements corresponding to confidence value at each step. 
+
+    Raises:
+        AttributeError: if metric type is not enum of Metrics.INSERTION or Metrics.DELETION
+    """
+    return _metric_calculation(attrs, transformed_img, model, chosen_class, metric_type=Metrics.INSERTION)
+    
+def visualize_metric(importance_lst:List[np.ndarray], metric_result:float, metric_type:str="Deletion"):
```

Comment:
Maybe we should move this function to `visualizer.py` file. What do you think?

Reply:
Fixed